### ISSUE TITLE
Pre Trained LSTM output clarification

### ISSUE THREAD

**Issue by mikedevelops** [adeshpande3/LSTM-Sentiment-Analysis#23]
Hey Adit, firstly, thank you for the article, this has been extremely useful in my research into sentiment analysis.

Would you mind clarifying 1 point for me, apologies if this is something obvious that I've missed.

In the pre-trained notebook the output from the `sess.run` is ...

```
[[-3.4586341   9.232184  ]
 [-2.0720391  -4.6618977 ]
 [ 1.36504     0.50249183]
 [ 2.5868893  -0.28297684]
 [-0.7046647  -3.8239028 ]
...]
```

Would you be able to explain how these numbers relate to the final sentiment analysis. I understand that the higher of the two numbers is the predicted sentiment, however I'm interested to know what the ranges of these numbers are and how we can use them to reason further about the sentiment.

I was expecting both values to be between `0` - `1`, and we could therefore deduce a percentage of the probable predicted sentiment. As this isn't the case, is there any light you can shed on the significance of the value of these numbers?

**Comment by adeshpande3** [comment]
Good question. So you can think of each of those inner lists as the prediction for each of the input strings in your test batch. Each of those inner lists has two numbers, the first of which represents negative sentiment score, and the other represents the positive sentiment score (or vice versa, can't remember off the top of my head) for that particular example. These numbers don't have a particular range because when we look at how we compute the prediction output,

![image](https://user-images.githubusercontent.com/13543909/36487441-bcaf9d02-16d5-11e8-91d4-629f00e2796f.png)

we see that the prediction value is in its logits form (aka unnormalized form). To convert the values into a probability between 0 and 1, we'd need to feed it into a [softmax layer](https://en.wikipedia.org/wiki/Softmax_function) which performs a form of normalization.

If you'd like to see that in the code, then you should add the following line after prediction
`normalizedPrediction = tf.nn.softmax(prediction)`  
and then change the sess.run call to 
`predictedSentiment = sess.run(normalizedPrediction, {input_data: inputMatrix})[0]`

I think that should work, but I'm a little low on time, so haven't gotten to try for myself yet. Lmk if you come across any issues.

**Comment by mikedevelops** [comment]
Thank you for getting back to me. That is great! Exactly what I was looking for.

I'm looking at Tensorflow Serving at the moment so having to really dig into the article and understand each part of the graph. My mind is a little blown at the moment, but I'm hoping that clears up soon 😄. Thank you for the help, much appreciated!
